src/Jackalope/Node.php

```diff
@@ -1639,7 +1665,8 @@ public function setChildrenDirty()
     public function setDeleted()
     {
         parent::setDeleted();
-        foreach ($this->properties as $property) {
+        foreach ($this->propertyValues as $name => $propertyValue) {
+            $property = $this->loadProperty($name);
```

Comment:
same here, only set existing property objects to deleted. all other objects did not exist so there can be no references to them.